src/crypto/CryptStateOCB2.cpp

```diff
@@ -55,20 +50,48 @@ void CryptState::genKey() {
 	bInit = true;
 }
 
-void CryptState::setKey(const unsigned char *rkey, const unsigned char *eiv, const unsigned char *div) {
-	memcpy(raw_key, rkey, AES_KEY_SIZE_BYTES);
-	memcpy(encrypt_iv, eiv, AES_BLOCK_SIZE);
-	memcpy(decrypt_iv, div, AES_BLOCK_SIZE);
-	AES_set_encrypt_key(raw_key, AES_KEY_SIZE_BITS, &encrypt_key);
-	AES_set_decrypt_key(raw_key, AES_KEY_SIZE_BITS, &decrypt_key);
-	bInit = true;
+void CryptStateOCB2::setKey(const std::string& rkey, const std::string& eiv, const std::string& div) {
+	if(rkey.length() == AES_KEY_SIZE_BYTES && eiv.length() == AES_BLOCK_SIZE && div.length() == AES_BLOCK_SIZE) {
+		memcpy(raw_key, rkey.data(), AES_KEY_SIZE_BYTES);
+		memcpy(encrypt_iv, eiv.data(), AES_BLOCK_SIZE);
+		memcpy(decrypt_iv, div.data(), AES_BLOCK_SIZE);
+		AES_set_encrypt_key(raw_key, AES_KEY_SIZE_BITS, &encrypt_key);
+		AES_set_decrypt_key(raw_key, AES_KEY_SIZE_BITS, &decrypt_key);
+		bInit = true;
+	}
+}
+
+void CryptStateOCB2::setRawKey(const std::string& rkey) {
+	if(rkey.length() == AES_KEY_SIZE_BYTES){
```

Comment:
If this condition is not met, the code at least log an error and maybe even completely fail (crash) the application. This seems to be a pre-condition